server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express'
+import url from 'url'
+import decisionTrees from '../../offenceCodeDecisions/DecisionTrees'
+import { FormError } from '../../@types/template'
+import validateForm from './offenceCodeDecisionsValidation'
+import PlaceOnReportService from '../../services/placeOnReportService'
+
+import type IncidentRole from '../../offenceCodeDecisions/IncidentRoles'
+import { properCaseName } from '../../utils/utils'
+
+type PageData = {
+  error?: FormError
+  selectedDecisionId?: string
+}
+
+export default class OffenceCodeRoutes {
+  constructor(private readonly placeOnReportService: PlaceOnReportService) {}
+
+  private async placeholderValues(adjudicationNumber: string, res: Response) {
+    const { user } = res.locals
+    const draftAdjudication = await this.placeOnReportService.getDraftAdjudicationDetails(
+      Number(adjudicationNumber),
+      user
+    )
+    const prisonerDetails = await this.placeOnReportService.getPrisonerDetails(
+      draftAdjudication.draftAdjudication.prisonerNumber,
+      user
+    )
+    return {
+      offenderFirstName: properCaseName(prisonerDetails.firstName),
+      offenderLastName: properCaseName(prisonerDetails.lastName),
+    }
+  }
+
+  private renderView = async (req: Request, res: Response, pageData?: PageData): Promise<void> => {
+    const { adjudicationNumber, incidentRole } = req.params
+    const { error, selectedDecisionId } = pageData
+    const placeholderValues = await this.placeholderValues(adjudicationNumber, res)
+    const path = req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, '')
```

Comment:
we grab everything after the incidentRole (which I've just hard coded for the time being and will need to be consistent with the rest of the application) and use it to find the decision in question